src/org/benf/cfr/reader/bytecode/analysis/parse/expression/ArithmeticMonOperation.java

```diff
@@ -58,6 +62,83 @@ public Dumper dumpInner(Dumper d) {
         return d;
     }
 
+    @Override
+    public Literal getComputedLiteral(Map<LValue, Literal> display) {
+        Literal l = lhs.getComputedLiteral(display);
+        if (l == null || !(l.getValue().getValue() instanceof Number))
+            return null;
+        Number lv = ((Number) l.value.getValue());
+        switch (l.getInferredJavaType().getRawType()) {
+            case BOOLEAN:
+            case BYTE:
+            case CHAR:
+            case SHORT:
+            case INT:
+                return getComputedInt(lv.intValue());
+            case LONG:
+                return getComputedLong(lv.longValue());
```

Comment:
Also making a note of here and in the other class to move the computation to a new class to hide some of the repetitive code in these expression classes.